src/Miki/Services/CommandTreeService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Miki.Framework.Commands;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Miki.Services
+{
+    public class CommandTreeService
+    {
+        private readonly CommandTree commandTree;
+
+        public CommandTreeService(CommandTree commandTree)
+        {
+            this.commandTree = commandTree;
+        }
+
+        private Node GetCommandFromSpecificModule(NodeModule module, string commandName)
+        {
+            commandName = commandName.ToLowerInvariant();
+
+            var commandNode = module.Children
+                            .FirstOrDefault(x => x.Metadata.Identifiers
+                            .Any(z => z.ToLowerInvariant() == commandName));
+
+            if(commandNode == null)
+            {
+                return null;
+            }
+
+            return commandNode;
+        }
+
+        public Node GetCommandByName(string commandName)
+        {
+            commandName = commandName.ToLowerInvariant();
+
+            var moduleNode = commandTree.Root.Children
+                    .OfType<NodeModule>()
+                    .FirstOrDefault(x => x.Children
+                        .Any(z => z.Metadata.Identifiers
+                            .Any(y=>y.ToLowerInvariant() == commandName)
+                        )
+                    );
+
+            if(moduleNode == null)
+            {
+                return null;
```

Comment:
perhaps a good idea to `throw EntityNullException<Node>("entity_command");` to make sure the return type is localized to the user.